### ISSUE TITLE
Supported browsers

### ISSUE THREAD

**Issue by Burtan** [dellisd/sqldelight-sqlite-wasm#3]
Maybe we could have a list of working browsers, for me the following work.

Linux (debian 12):
Chromium (111.0)
Google Chrome (111.0)
Firefox (111.0)

Android (12):
Google Chrome (111.0)
Firefox (111.0)

Windows (10).
Google Chrome (111.0)
Firefox (111.0)
Edge (111.0)

MacOS (13.2.1):
Safari does not work! Although support of OPFS is there (https://webkit.org/blog/12257/the-file-system-access-api-with-origin-private-file-system/)

iOS (16.3.1)
Safari does not work.

Basically all major systems and browsers seem to work in their current versions but Apple's. Here seems to be some discussion on it on the site of sqlite: https://sqlite.org/forum/forumpost/3759f37a7c3e7e52217a594f4a7188f21e9e4f3a0192e0a5658235745c6f0569

Let's see how long Apple keeps Safari back to protect its AppStore...

**Comment by Burtan** [comment]
It seems that Safari does not work because of problems with WebWorkers being started inside of a WebWorker and not because of missing OPFS support. While current Safari does not even allow to create WebWorkers inside of WebWorkers, the Safari tech preview allows to create them, but somehow they are not started, at least not in sqlite3.js.

The relevant line is 10842: 
`new Worker(new URL("sqlite3-opfs-async-proxy.js", import.meta.url));`
The worker never starts and therefore the promise, where the worker is created is never resolved in its onmessage function.

**Comment by dellisd** [comment]
I'm a bit confused by what you mean about a web worker being started "inside" of another web worker? I don't see where that's happening here.

I am surprised that Safari isn't able to load the worker in this case though.

**Comment by Burtan** [comment]
Our web worker [sqlite.worker.js](https://github.com/dellisd/sqldelight-sqlite-wasm/blob/main/src/jsMain/resources/sqlite.worker.js) loads sqlite3.js (from the sqlite wasm package), which in turn starts a web worker (line 10859) sqlite3-opfs-async-proxy.js (also from the sqlite wasm package).

**Comment by Burtan** [comment]
By using https://github.com/dmihal/Subworkers I could work around safaris incapability of nested workers. However, there is still an error thrown. I`ll look further into it.

**Comment by dellisd** [comment]
Ah! Interesting. I didn't realize sqlite-wasm was starting an additional worker. I'm going to review the sqlite docs about it, but I feel like if it's possible I'd like to avoid creating the second worker altogether and have sqlite run in the main driver worker instead.

**Comment by Burtan** [comment]
I created a [bug report for webkit](https://bugs.webkit.org/show_bug.cgi?id=254609) if anybody is interested.

**Comment by Burtan** [comment]
OPFS is working on safari 17
